### PR TITLE
Honor cms.linkPolicy config in ImageResizer

### PR DESCRIPTION
**Pull request by mjauvin**

### CODE REVIEW

**Review by bennothommo — APPROVED** [review]

**Comment by LukeTowers** [comment]
@mjauvin does this support being installed in a subfolder? Or do we require linkPolicy to be force in those cases?

**Comment by mjauvin** [comment]
@LukeTowers I guess not, but wouldn't it be the case here as well?

ref. https://github.com/octobercms/october/blob/develop/modules/system/classes/MediaLibrary.php#L541-L552

I think we need to revisit the paths for the whole stack regarding subfolder installations, don't you think?

**Comment by LukeTowers** [comment]
@mjauvin just checked the config file and I think we're covered for subfolder use because we clearly document how to make sure that works: https://github.com/octobercms/october/blob/develop/config/cms.php#L303-L306

File: modules/system/classes/ImageResizer.php [hunk 1]
```diff
@@ -499,7 +499,13 @@ public function getResizerUrl()
         // Store the current configuration
         $this->storeConfig();
 
-        return Url::to("/resizer/$identifier/$resizedUrl");
+        $path = "/resizer/$identifier/$resizedUrl";
+
+        if (Config::get('cms.linkPolicy') === 'force') {
```
Comment (LukeTowers) [inline modules/system/classes/ImageResizer.php]:
```suggestion
        if (Config::get('cms.linkPolicy', 'detect') === 'force') {
```

File: modules/system/classes/ImageResizer.php [hunk 2]
```diff
@@ -523,7 +529,13 @@ public function getResizedUrl()
         // Ensure that a properly encoded URL is returned
         $segments = explode('/', $url);
         $lastSegment = array_pop($segments);
-        return implode('/', $segments) . '/' . rawurlencode(rawurldecode($lastSegment));
+        $path = implode('/', $segments) . '/' . rawurlencode(rawurldecode($lastSegment));
+
+        if (Config::get('cms.linkPolicy') === 'force') {
```
Comment (LukeTowers) [inline modules/system/classes/ImageResizer.php]:
```suggestion
        if (Config::get('cms.linkPolicy', 'detect') === 'force') {
```

File: modules/system/classes/ImageResizer.php [hunk 3]
```diff
@@ -499,7 +499,13 @@ public function getResizerUrl()
...
         $this->storeConfig();
 
-        return Url::to("/resizer/$identifier/$resizedUrl");
+        $path = "/resizer/$identifier/$resizedUrl";
+
+        if (Config::get('cms.linkPolicy', 'detect') === 'force') {
+            return Url::to($path);
+        }
+
+        return $path;
```
Comment (LukeTowers) [inline modules/system/classes/ImageResizer.php]:
```suggestion
        $url = "/resizer/$identifier/$resizedUrl";

        if (Config::get('cms.linkPolicy', 'detect') === 'force') {
            $url = Url::to($url);
        }

        return $url;
```

File: modules/system/classes/ImageResizer.php [hunk 4]
```diff
@@ -523,7 +529,13 @@ public function getResizedUrl()
...
         $segments = explode('/', $url);
         $lastSegment = array_pop($segments);
-        return implode('/', $segments) . '/' . rawurlencode(rawurldecode($lastSegment));
+        $path = implode('/', $segments) . '/' . rawurlencode(rawurldecode($lastSegment));
+
+        if (Config::get('cms.linkPolicy', 'detect') === 'force') {
+            return Url::to($path);
+        }
+
+        return $path;
```
Comment (LukeTowers) [inline modules/system/classes/ImageResizer.php]:
```suggestion
        $url = implode('/', $segments) . '/' . rawurlencode(rawurldecode($lastSegment));

        if (Config::get('cms.linkPolicy', 'detect') === 'force') {
            $url = Url::to($url);
        }

        return $url;
```

**Comment by LukeTowers** [comment]
@mjauvin could you add some unit tests for the linkPolicy settings?

**Comment by mjauvin** [comment]
But the path here is static in the code to "/resizer/..." this is not configurable.

**Comment by LukeTowers** [comment]
@mjauvin I'm not sure what you mean?

**Comment by mjauvin** [comment]
there should be a config for the resizer path prefix, it's currently hard-coded to "/resizer"

**Comment by LukeTowers** [comment]
I don't see why that should be made configurable, it's a core piece of functionality handled by October. We don't make the combiner URL configurable: https://github.com/octobercms/october/blob/develop/modules/system/routes.php#L15

**Comment by mjauvin** [comment]
Ok, so even when in a sub-folder installation you're saying this won't cause problems? Fine with me.

**Comment by mjauvin** [comment]
unit tests for the linkPolicy settings, do you mean for `getResizerUrl()` ?

**Comment by mjauvin** [comment]
Odd, we got "Segmentation fault" for php7.2 & 7.3 tests...

**Comment by LukeTowers** [comment]
And for getResizedUrl(), just to verify that those methods are generating URLs according to the linkPolicy set. Don't worry about the failing tests for now, seems like it's a github issue.

**Comment by mjauvin** [comment]
tests added @LukeTowers !

**Comment by mjauvin** [comment]
ping

**Comment by LukeTowers** [comment]
What's wrong with the tests @mjauvin?

**Comment by mjauvin** [comment]
@LukeTowers it's one of those transient unexplained failures.

**Comment by LukeTowers** [comment]
@mjauvin can you push another commit (perhaps merging develop into this branch?). I don't seem to be able to re-run the checks manually

**Comment by LukeTowers** [comment]
Thanks @mjauvin, good job!

**Pull request merged**